decode.go

```diff
@@ -0,0 +1,60 @@
+package iso8583
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// Unmarshal traverses the message fields recursively and for each field, sets
+// the field value into the corresponding struct field value pointed by v. If v
+// is nil or not a pointer it returns error.
+func Unmarshal(message *Message, v interface{}) error {
+	rv := reflect.ValueOf(v)
+	if rv.Kind() != reflect.Ptr || rv.IsNil() {
+		return errors.New("data is not a pointer or nil")
+	}
+
+	// get the struct from the pointer
+	dataStruct := rv.Elem()
+
+	if dataStruct.Kind() != reflect.Struct {
+		return errors.New("data is not a struct")
+	}
+
+	// iterate over struct fields
+	for i := 0; i < dataStruct.NumField(); i++ {
+		dataFieldName := dataStruct.Type().Field(i).Name
+
+		// skip struct field if its name starts not from F
+		if len(dataFieldName) == 0 || dataFieldName[0:1] != "F" {
+			continue
+		}
+
+		indexStr := dataFieldName[1:]
+		fieldIndex, err := strconv.Atoi(indexStr)
+		if err != nil {
+			return errors.Wrap(err, "converting field intex into int")
```

Comment:
Still need to replace this with `fmt.Errorf`.

We can import the stdlib `errors` instead of `pkg/errors` and still have `errors.New` functionality, without adding a new dependency.

Reply:
Oh yeah. Fixed it! Thanks!